x/market/types/msgs.go

```diff
@@ -12,9 +12,10 @@ const (
 
 // MsgCreateBid defines an SDK message for creating Bid
 type MsgCreateBid struct {
-	Order    OrderID        `json:"order"`
-	Provider sdk.AccAddress `json:"owner"`
-	Price    sdk.Coin       `json:"price"`
+	Order         OrderID         `json:"order"`
+	Provider      sdk.AccAddress  `json:"owner"`
+	Price         sdk.Coin        `json:"price"`
+	RequiredAttrs []sdk.Attribute `json:"attributes"`
```

Comment:
this shouldn't be here.  the provider just chooses to not bid if these new attributes don't match.

again, it's similar to the price being lower than what the provider is willing to bid on.